pkg/operator/controller/controller_router_deployment.go

```diff
@@ -290,8 +303,9 @@ func deploymentConfigChanged(current, expected *appsv1.Deployment) (bool, *appsv
 	return true, updated
 }
 
-func cmpEnvs(a, b corev1.EnvVar) bool    { return a.Name < b.Name }
-func cmpVolumes(a, b corev1.Volume) bool { return a.Name < b.Name }
+func cmpEnvs(a, b corev1.EnvVar) bool            { return a.Name < b.Name }
+func cmpTolerations(a, b corev1.Toleration) bool { return a.Key < b.Key }
```

Comment:
I think we need to compare all the fields because you could have two tolerations with the matching `Key` but non-matching other fields.  For example, as far as I can tell, it is possible (even if silly) to have a toleration with key `x` and operator `Exists` and another toleration with the same key but with operator `Equals`, or you could have two tolerations with the same key and operator but different effects.

Reply:
Done.

Reply:
Done.